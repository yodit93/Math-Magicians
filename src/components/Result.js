import PropTypes from 'prop-types';

const Result = ({ state }) => {
  const { total, next, operation } = state;
  return (
    <div className="result-cont" data-testid="test">
      <span className="result">
        {total}
        {' '}
        {operation}
        {' '}
        {next}
      </span>
    </div>
  );
};

Result.propTypes = {
  state: PropTypes.objectOf(PropTypes.string).isRequired,
};

export default Result;
