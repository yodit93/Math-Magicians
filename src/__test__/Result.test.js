import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Result from '../components/Result';

const state = {
  title: '3',
  next: '4',
  operation: '+',
};

test('should render the Result component', () => {
  const tree = renderer.create(<Result state={state} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should have two paragraphs', () => {
  render(<Result state={state} />);
  const divElement = screen.getByTestId('test');
  expect(divElement).toBeInTheDocument();
});
