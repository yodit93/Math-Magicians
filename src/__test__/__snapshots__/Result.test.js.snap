// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Result component 1`] = `
<div
  className="result-cont"
  data-testid="test"
>
  <span
    className="result"
  >
     
    +
     
    4
  </span>
</div>
`;
